src/Symfony/Component/Cache/Traits/Redis5Proxy.php

```diff
@@ -0,0 +1,1234 @@
+    {
+        return $this->lazyObjectReal->lSize(...\func_get_args());
+    }
+
+    public function listTrim($key, $start, $stop)
+    {
+        return $this->lazyObjectReal->listTrim(...\func_get_args());
+    }
+
+    public function open($host, $port = null, $timeout = null, $retry_interval = null)
+    {
+        return $this->lazyObjectReal->open(...\func_get_args());
+    }
+
+    public function popen($host, $port = null, $timeout = null)
+    {
+        return $this->lazyObjectReal->popen(...\func_get_args());
+    }
+
+    public function renameKey($key, $newkey)
+    {
+        return $this->lazyObjectReal->renameKey(...\func_get_args());
+    }
+
+    public function sContains($key, $value)
+    {
+        return $this->lazyObjectReal->sContains(...\func_get_args());
+    }
+
+    public function sGetMembers($key)
+    {
+        return $this->lazyObjectReal->sGetMembers(...\func_get_args());
+    }
+
+    public function sRemove($key, $member, ...$other_members)
+    {
+        return $this->lazyObjectReal->sRemove(...\func_get_args());
+    }
+
+    public function sSize($key)
+    {
+        return $this->lazyObjectReal->sSize(...\func_get_args());
+    }
+
+    public function sendEcho($msg)
+    {
+        return $this->lazyObjectReal->sendEcho(...\func_get_args());
+    }
+
+    public function setTimeout($key, $timeout)
+    {
+        return $this->lazyObjectReal->setTimeout(...\func_get_args());
+    }
+
+    public function substr($key, $start, $end)
+    {
+        return $this->lazyObjectReal->substr(...\func_get_args());
+    }
+
+    public function zDelete($key, $member, ...$other_members)
+    {
+        return $this->lazyObjectReal->zDelete(...\func_get_args());
+    }
+
+    public function zDeleteRangeByRank($key, $min, $max)
+    {
+        return $this->lazyObjectReal->zDeleteRangeByRank(...\func_get_args());
+    }
+
+    public function zDeleteRangeByScore($key, $min, $max)
+    {
+        return $this->lazyObjectReal->zDeleteRangeByScore(...\func_get_args());
+    }
+
+    public function zInter($key, array $keys, ?array $weights = null, $aggregate = null)
+    {
+        return $this->lazyObjectReal->zInter(...\func_get_args());
+    }
+
+    public function zRemove($key, $member, ...$other_members)
+    {
+        return $this->lazyObjectReal->zRemove(...\func_get_args());
+    }
+
+    public function zRemoveRangeByScore($key, $min, $max)
+    {
+        return $this->lazyObjectReal->zRemoveRangeByScore(...\func_get_args());
+    }
+
+    public function zReverseRange($key, $start, $end, $scores = null)
+    {
+        return $this->lazyObjectReal->zReverseRange(...\func_get_args());
+    }
+
+    public function zSize($key)
+    {
+        return $this->lazyObjectReal->zSize(...\func_get_args());
+    }
+
+    public function zUnion($key, array $keys, ?array $weights = null, $aggregate = null)
+    {
+        return $this->lazyObjectReal->zUnion(...\func_get_args());
+    }
```

Comment:
Just wondering why `__call()` and `__callStatic()` methods have not been added to the `LazyProxyTrait`?

Reply:
Because it's not part of the base Redis class, so it wouldn't serve any purpose. Or did I miss something in your question?

Reply:
I mean using `__call()` instead of re creating all the methods one by one.

Reply:
That's what we do right now, but doing so means our RedisProxy cannot extend the base Redis, which means not passing `instanceof Redis` checks.